Adapter.C

```diff
@@ -923,6 +924,69 @@ void preciceAdapter::Adapter::setupCheckpointing()
+    {                                                                                                                 \
+        regFields.push_back(obj);                                                                                     \
+    }                                                                                                                 \
+    /* Iterate through checkpointed fields */                                                                         \
+    for (uint i = 0; i < GeomFieldCopies_.size(); i++)                                                                \
+    {                                                                                                                 \
+        found = false;                                                                                                \
+        obj = GeomFieldCopies_.at(i)->name();                                                                         \
+        for (uint j = 0; j < regFields.size(); j++)                                                                   \
+        {                                                                                                             \
+            if (obj == regFields.at(j))                                                                               \
+            {                                                                                                         \
+                found = true;                                                                                         \
+                break;                                                                                                \
+            }                                                                                                         \
+        }                                                                                                             \
+        if (!found)                                                                                                   \
+        {                                                                                                             \
+            toRemove.push_back(obj);                                                                                  \
+        }                                                                                                             \
+    }                                                                                                                 \
+    if (!toRemove.empty())                                                                                            \
+    {                                                                                                                 \
+        for (int i = toRemove.size() - 1; i >= 0; i--)                                                                \
```

Comment:
You explained to me why you chose to iterate in reverse order here. I think that would be a useful comment to add.